StartEycaActivation/handler.ts

```diff
@@ -0,0 +1,270 @@
+  );
+
+export function StartEycaActivationHandler(
+  userEycaCardModel: UserEycaCardModel,
+  userCgnModel: UserCgnModel,
+  logPrefix: string = "StartEycaActivationHandler"
+): IStartCgnActivationHandler {
+  return async (context, fiscalCode) => {
+    const client = df.getClient(context);
+    const orchestratorId = makeEycaOrchestratorId(
+      fiscalCode,
+      PendingStatusEnum.PENDING
+    ) as NonEmptyString;
+
+    const isEycaEligibleOrError = await getEycaEligibleTask(
+      fiscalCode,
+      userCgnModel
+    ).run();
+    if (isLeft(isEycaEligibleOrError)) {
+      return isEycaEligibleOrError.value;
+    }
+
+    const card: CardPending = {
+      status: PendingStatusEnum.PENDING
+    };
+
+    return userEycaCardModel
+      .findLastVersionByModelId([fiscalCode])
+      .mapLeft<ErrorTypes | IResponseSuccessAccepted>(() =>
+        ResponseErrorInternal("Cannot query EYCA data")
+      )
+      .chain(maybeUserEycaCard =>
+        maybeUserEycaCard.foldL(
+          () => taskEither.of(fiscalCode),
+          userEycaCard =>
+            // if an EYCA card is already in a final state we return Conflict
+            [
+              ActivatedStatusEnum.ACTIVATED.toString(),
+              ExpiredStatusEnum.EXPIRED.toString(),
+              RevokedStatusEnum.REVOKED.toString()
+            ].includes(userEycaCard.card.status)
+              ? fromLeft(
+                  ResponseErrorConflict(
+                    `Cannot activate an EYCA card that is already ${userEycaCard.card.status}`
+                  )
+                )
+              : // if EYCA card is in PENDING status, try to get orchestrator status
+                // in order to discriminate if there's an error or not
+                tryCatch(() => client.getStatus(orchestratorId), toError)
+                  .mapLeft<ErrorTypes | IResponseSuccessAccepted>(() =>
+                    ResponseErrorInternal("Cannot retrieve activation status")
+                  )
+                  .chain(maybeStatus =>
+                    // client getStatus could respond with undefined if
+                    // an orchestrator instance does not exists
+                    // see https://docs.microsoft.com/it-it/azure/azure-functions/durable/durable-functions-instance-management?tabs=javascript#query-instances
+                    fromNullable(maybeStatus).foldL(
+                      // if orchestrator does not exists we assume that it expires its storage in TaskHub
+                      // after 30 days so we can try to start a new activation process
+                      () => taskEither.of(fiscalCode),
+                      _ =>
+                        // if orchestrator is running we return an Accepted Response
+                        // otherwise we assume the orchestrator is in error or
+                        // it has been canceled so we can try to start a new activation process
+                        mapOrchestratorStatus(_).map(() => fiscalCode)
```

Comment:
We are ignoring [here](https://github.com/pagopa/io-functions-cgn/pull/20/files#diff-70158054dede1408e047baec50926bfe9d3c6117eff746175f11efa98555ffa8R192) the right value of the chain (always `fiscalcode`), can we pass `void 0` instead to avoid confusion?